community/cloud-foundation/templates/app_engine/tests/integration/app_engine.bats

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env bats
+
+source tests/helpers.bash
+
+TEST_NAME=$(basename "${BATS_TEST_FILENAME}" | cut -d '.' -f 1)
+
+# Create a random 10-char string and save it in a file.
+RANDOM_FILE="/tmp/${CLOUD_FOUNDATION_ORGANIZATION_ID}-${TEST_NAME}.txt"
+if [[ ! -e "${RANDOM_FILE}" ]]; then
+    RAND=$(head /dev/urandom | LC_ALL=C tr -dc a-z0-9 | head -c 10)
+    echo ${RAND} > "${RANDOM_FILE}"
+fi
+
+# Set variables based on the random string saved in the file.
+# envsubst requires all variables used in the example/config to be exported.
+if [[ -e "${RANDOM_FILE}" ]]; then
+    export RAND=$(cat "${RANDOM_FILE}")
+    DEPLOYMENT_NAME="${CLOUD_FOUNDATION_PROJECT_ID}-${TEST_NAME}-${RAND}"
+    # Replace underscores in the deployment name with dashes.
+    DEPLOYMENT_NAME=${DEPLOYMENT_NAME//_/-}
+    CONFIG=".${DEPLOYMENT_NAME}.yaml"
+fi
+
+########## HELPER FUNCTIONS ##########
+
+function create_config() {
+    echo "Creating ${CONFIG}"
+    envsubst < ${BATS_TEST_DIRNAME}/${TEST_NAME}.yaml > "${CONFIG}"
+}
+
+function delete_config() {
+    echo "Deleting ${CONFIG}"
+    rm -f "${CONFIG}"
+}
+
+function setup() {
+    # Global setup; executed once per test file.
+    if [ ${BATS_TEST_NUMBER} -eq 1 ]; then
+        create_config
+    fi
+
+    # Per-test setup steps.
+}
+
+function teardown() {
+    # Global teardown; executed once per test file.
+    if [[ "$BATS_TEST_NUMBER" -eq "${#BATS_TEST_NAMES[@]}" ]]; then
+        delete_config
+        rm -f "${RANDOM_FILE}"
+    fi
+
+    # Per-test teardown steps.
+}
+
+@test "Creating deployment ${DEPLOYMENT_NAME} from ${CONFIG}" {
+    run gcloud deployment-manager deployments create "${DEPLOYMENT_NAME}" \
+        --config "${CONFIG}" --project "${CLOUD_FOUNDATION_PROJECT_ID}"
+    [[ "$status" -eq 0 ]]
+}
+
+@test "Verifying resource was created in deployment ${DEPLOYMENT_NAME}" {
+    run gcloud app describe --project "${CLOUD_FOUNDATION_PROJECT_ID}"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "id: ${CLOUD_FOUNDATION_PROJECT_ID}" ]]
+    [[ "$output" =~ "locationId: us-east1" ]]
+    [[ "$output" =~ "servingStatus: SERVING" ]]
+}
+
+@test "Verifying that flexible app engine resource were created in deployment ${DEPLOYMENT_NAME}" {
```

Comment:
Can we also add a standard version to the same app engine to test standard is being created ok?